### ISSUE TITLE
Windows Conntrack:  can not reverse DNAT as expected

### ISSUE THREAD

**Issue by ruicao93** [openvswitch/ovs-issues#190]
Hi,

We got a issue when using DNAT function on windows.

We do DNAT for the request pkts for a connection, but found that the souce IP of reply packte is not reversed as expected.

Take following  connection as example:
```
tcp,orig=(src=192.168.187.214,dst=10.108.85.76,sport=50207,dport=80),reply=(src=192.168.187.214,dst=192.168.187.214,sport=8080,dport=50207),zone=65519,protoinfo=(state=SYN_SENT)
```
- Request packet:
  (src=192.168.187.214,dst=10.108.85.76,sport=50207,dport=80)
- Request packet after DNAT(dst=192.168.187.214:8080)
  (src=192.168.187.214,dst=192.168.187.214,sport=50207,dport=8080)
- Reply packet before reverse DNAT:
  (src=192.168.187.214,dst=192.168.187.214,sport=8080,dport=50207)
- Reply packet after reverse DNAT:
  Actual: (src=192.168.187.214,dst=192.168.187.214,sport=8080,dport=50207)
  Expected: (src=10.108.85.76, dst=192.168.187.214,sport=80,dport=50207)

Anyone could help debug the issue? I used the latest code on branch-2.13

Thanks,
Rui

**Comment by commandgjj** [comment]
I analyzed this issue. The root cause is the flow key is not updated in OVS_ACTION_ATTR_SET operation. It will impact follow-up  actions, such as, conntrack module which cannot find created entry if passing old flow key.

I have cooked the patch and will post PR soon. Thanks!

**Comment by commandgjj** [comment]
@aserdean 

The PR: https://github.com/openvswitch/ovs/pull/326

The patch also need merged into 2.13 and 2.14. Thanks!

**Comment by ruicao93** [comment]
> @aserdean
> 
> The PR: [openvswitch/ovs#326](https://github.com/openvswitch/ovs/pull/326)
> 
> The patch also need merged into 2.13 and 2.14. Thanks!

Thanks @commandgjj 
@aserdean , @williamtu , could you help port the fix to 2.13 and 2.14? 

Thanks,
Rui

**Comment by aserdean** [comment]
I applied the https://github.com/openvswitch/ovs/commit/ba73001b6e1d1048df149d7d024e5fecd8ca618d as far as branch-2.8.

I'm closing this issue. Please feel free to open it if it reproduces.
